### ISSUE TITLE
when predict online is so slowly

### ISSUE THREAD

**Issue by rickyhuw** [microsoft/LightGBM#3741]
when I reload the 'model.txt' online for predict ranking ,the time over 300ms, how to solve the problem?

**Comment by jameslamb** [comment]
Can you please provide more details?

* how are you using LightGBM (CLI, Python, R, something else)
* what parameters did you use for training? This will tell us how large the model is.
* what do you mean by "predict online"? Are you re-loading the model each time you generate a prediction?

**Comment by guolinke** [comment]
@rickyhuw LightGBM is mainly optimized for training. For inference, you can convert the model to cpp code (refer to https://lightgbm.readthedocs.io/en/latest/Parameters.html#convert-parameters). 
you can also use other libraries, such as treelite, to load LightGBM model.

**Comment by rickyhuw** [comment]
> Can you please provide more details?
> 
> * how are you using LightGBM (CLI, Python, R, something else)?
> * what parameters did you use for training? This will tell us how large the model is.
> * what do you mean by "predict online"? Are you re-loading the model each time you generate a prediction?

The Details:
* Python is used and the service is deployed with the flash framework
* parameters:params = {
    'task': 'train',
    'nthread': -1,
    'boosting_type': 'gbdt',
    'objective': 'binary',
    'metric': {'binary_logloss', 'auc'},
    'num_leaves': 31,
    'num_trees': 400,
    'learning_rate': 0.08,
    'feature_fraction': 0.9,
    'bagging_fraction': 0.8,
    'bagging_freq': 3,
    'verbose': 1,
    'max_depth': 10,
    'lambda_l1': 0.001,
    'lambda_l2': 0.001,
    'min_data_in_leaf': 100,
    'max_bin':300,
    'force_col_wise': True
}
* Using model parameters to provide online inference service, the "init function" used will only load the model once
**Hope to get a reply, thank you**

**Comment by StrikerRUS** [comment]
@rickyhuw Please see this answer https://github.com/microsoft/LightGBM/issues/3741#issuecomment-757482701.

Also you may find this comment useful https://github.com/microsoft/LightGBM/issues/2397#issuecomment-530340871 (treelite has Python runtime API for your use-case with Flask).
https://treelite.readthedocs.io/en/latest/treelite-runtime-api.html
